cmd/snap-confine/seccomp-support.c

```diff
@@ -152,24 +240,22 @@ bool sc_apply_seccomp_profile_for_security_tag(const char *security_tag)
 	// set on the system.
 	validate_bpfpath_is_safe(profile_path);
 
-    /* The extra space has dual purpose. First of all, it is required to detect
-     * feof() while still being able to correctly read MAX_BPF_SIZE bytes of
-     * seccomp profile.  In addition, because we treat the profile as a
-     * quasi-string and use sc_streq(), to compare it. The extra space is used
-     * as a way to ensure the result is a terminated string (though in practice
-     * it can contain embedded NULs any earlier position). Note that
-     * sc_read_seccomp_filter knows about the extra space and ensures that the
-     * buffer is never empty. */
-	char bpf[MAX_BPF_SIZE + 1] = { 0 };
-	size_t num_read = sc_read_seccomp_filter(profile_path, bpf, sizeof bpf);
-	if (sc_streq(bpf, "@unrestricted\n")) {
+	// TODO: strange init to workaround bug in gcc on 14.04 that claims:
+	//  "error: missing braces around initializer"
+	struct sc_seccomp_file_header hdr = {{0},0};
```

Comment:
Does this produce the error also? This has the same effect
```suggestion
	struct sc_seccomp_file_header hdr = {{0}};
```

Reply:
Nice, your suggestion seems to work fine and is indeed a bit easier to read. I also tweaked the comment and removed the TODO